sql-virtual-machine/register-sql-vms/RegisterSqlVMs.psm1

```diff
@@ -712,16 +694,23 @@ function assert-Subscription(
     if ($Global:Error) {
         $connectionError = $Global:Error[0]
         $errorMessage = "$($Subscription), $($connectionError[0].Exception.Message)"
-        Write-Output $errorMessage | Out-File $Global:LogFile
+        Write-Output $errorMessage | Out-File $Global:LogFile -Append
         $tmp = $Global:SubscriptionsFailedToConnect.Add($Subscription)
         return $false
     }
 
     # register Subscription with SQL VM RP
     $registration = Get-AzResourceProvider -ProviderNamespace Microsoft.SqlVirtualMachine -ErrorAction SilentlyContinue
     if ((!$registration) -or ($registration[0].RegistrationState -ne 'Registered')) {
-        $errorMessage = "$($Subscription), Subscription $($Subscription) should be registered to 'Microsoft.SqlVirtualMachine'. This registration may take around 5 mins to propogate. Steps to register can be found here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-supported-services"
-        Write-Output $errorMessage | Out-File $Global:LogFile
+        # try register subscription to the SqlVirtualMachine
+        $register = Register-AzResourceProvider -ProviderNamespace Microsoft.SqlVirtualMachine -ErrorAction SilentlyContinue
+        if ((!$register) -or ($register.RegistrationState -ne 'Registering')) {
+            $errorMessage = "$($Subscription), Subscription $($Subscription) should be registered to 'Microsoft.SqlVirtualMachine'. Steps to register can be found here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-supported-services. This registration may take around 5 mins to propogate."
```

Comment:
Do not include locales in URLs. This forces the user to English. Change the URL to:

`https://docs.microsoft.com/azure/azure-resource-manager/resource-manager-supported-services`

The Docs platform detects the user's locale based on their browser configuration. This allows the Docs platform to redirect the user to localized content if available in their language.

Reply:
Thank you Sean. I have updated all occurrences by removing the locale. I have changed few typos as well